src/Extensions/TimedScopes/Internal/TimedScopeProvider.cs

```diff
@@ -0,0 +1,40 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT license.
+
+using System;
+using Microsoft.Omex.Extensions.Abstractions;
+
+namespace Microsoft.Omex.Extensions.TimedScopes
+{
+	internal sealed class TimedScopeProvider : ITimedScopeProvider
+	{
+		public TimedScopeProvider(
+			ITimedScopeEventSender eventSource,
+			IActivityProvider activityProvider,
+			ILogEventReplayer? logReplayer = null)
+		{
+			m_activityProvider = activityProvider;
+			m_eventSource = eventSource;
+			m_logReplayer = logReplayer;
+		}
+
+
+		public TimedScope Start(string name, TimedScopeResult result) => Create(name, result).Start();
+
+
+		public TimedScope Create(string name, TimedScopeResult result)
+		{
+			// Activity won't be stated in case of an empty string
+			if (string.IsNullOrWhiteSpace(name))
+			{
+				throw new ArgumentException("TimedScope name should not be empty or null");
+			}
```

Comment:
Skip line after `}` as per convention

Reply:
Changed implementation of the method to delegate validation, sorry, thank you for your comment